pkg/auth/handler/userverify/verify_request.go

```diff
@@ -124,20 +130,30 @@ func (h VerifyRequestHandler) Handle(req interface{}) (resp interface{}, err err
 	}
 	user := userFactory.NewUser(*authInfo, userProfile)
 
-	var value string
-	var ok bool
-	if value, ok = user.LoginIDs[payload.RecordKey]; !ok {
-		err = skyerr.NewError(skyerr.UnexpectedError, "Value of "+payload.RecordKey+" doesn't exist.")
+	principals, err := h.PasswordAuthProvider.GetPrincipalsByLoginID(payload.LoginIDKey, payload.LoginID)
```

Comment:
We do not consider realm here. Can we document the expected behavior here?

Reply:
This behavior is documented in proposed spec. I'll add a comment to clarify the intention here.